addon/components/flexberry-dropdown.js

```diff
@@ -165,12 +165,18 @@ export default FlexberryBaseComponent.extend({
       Ember.Logger.error(`Wrong type of flexberry-dropdown \`items\` propery: actual type is ${Ember.typeOf(items)}, but array is expected.`);
     }
 
-    var value = this.get('value') || null;
-    if (needChecksOnValue && !Ember.isNone(value) && items.indexOf(value) < 0) {
+    // Convert 'value' and 'items' to strings because flexberry-dropdown interpret selected value as string commonly.
+    let value = this.get('value') || null;
+    let stringValue = value ? value.toString() : null;
+    let itemsString = items.map(function(item) {
+      return item ? item.toString() : null;
+    });
+
+    if (needChecksOnValue && !Ember.isNone(stringValue) && itemsString.indexOf(stringValue) < 0) {
```

Comment:
Type conversion before check doesn't guarantee that the specified value matches one of the allowed items.
Are you sure tha it's impossible to force ui-dropdown to return selected item value without conversion?